doc/parameters.txt

```diff
@@ -153,14 +153,18 @@ Installer
     three answerfile options above, "" otherwise.
 
 
-  --map_netdev=eth:d|s:mac|pci[[index]]|ppn
-
-    Map interface with the given MAC address, PCI address or PPN to the given
-    interface name.
-    Some NICs expose multiple interfaces for a single PCI address.  In these
-    cases the PPN is often not reliable.  An index to a PCI address can be
-    specified if required (NICs are sorted in MAC order).  An index of 0 is
-    implied when one is not provided.
+  --map_netdev=[eth]X[:d|s]:mac|pci[[index]]|label
+
+    Map interface with the given MAC address, PCI address or label to the given interface name.
+    Some NICs expose multiple interfaces for a single PCI address
+    An index to a PCI address can be specified if required (NICs are sorted in MAC order).
+
+    - eth: literal string "eth", has no meaning, just for backward compatibility
+    - X: number, starting from 0,  the expected order of this interface
+    - d|s: has no meaning, just for backward compatibility
+    - mac: valid Mac address
+    - pci: valid PCI address. An index of 0 is implied when one is not provided
+    - label: valid interface name, it needs to be quoted to distinguish from other kind and take priority than others
```

Comment:
Does it really need to be quoted? A valid MAC address or PCI address would start with a number and include colons. I think a valid interface name would always start with a letter and not include any colons.

Passing quotes in from the bootloader can be a bit tricky since some bootloaders (like GRUB) interpret quotes themselves.

Reply:
Both MAC and PCI are hex, rather than decimal.  They can in principle be ambiguous with a name label, although filtering on a colon or not is almost certainly good enough.

Reply:
It is probably best to treat it as a label if it accepts quotes (for backwards compatibility) or if the string does not contain a colon.